cmake/modules/FindOmrCuda.cmake

```diff
@@ -0,0 +1,72 @@
+###############################################################################
+# Copyright (c) 2019, 2019 IBM Corp. and others
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which accompanies this
+# distribution and is available at http://eclipse.org/legal/epl-2.0
+# or the Apache License, Version 2.0 which accompanies this distribution
+# and is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# This Source Code may also be made available under the following Secondary
+# Licenses when the conditions for such availability set forth in the
+# Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+# version 2 with the GNU Classpath Exception [1] and GNU General Public
+# License, version 2 with the OpenJDK Assembly Exception [2].
+#
+# [1] https://www.gnu.org/software/classpath/license.html
+# [2] http://openjdk.java.net/legal/assembly-exception.html
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+#############################################################################
+
+#
+# This package locates a minimum set of CUDA resources, required by the OMR_OPT_CUDA flag.
+#
+
+include(FindPackageHandleStandardArgs)
+
+set(OMR_CUDA_HOME "NOTFOUND" CACHE STRING "Path to the CUDA SDK. Takes precedence over CUDA_HOME in OMR.")
+
+set(OmrCuda_INCLUDE_DIRS "OmrCuda_INCLUDE_DIRS-NOTFOUND")
+
+set(OmrCuda_SEARCH_PATH
+	"${OMR_CUDA_HOME}"
+	"$ENV{CUDA_HOME}"
+	"$ENV{CUDA_PATH}"
+	"$ENV{CUDA_INC_PATH}"
+	"$ENV{NVSDKCOMPUTE_ROOT}/C"
```

Comment:
Why the `/C` suffix here?

Reply:
Stolen from [FindCUDA.cmake](https://gitlab.kitware.com/cmake/cmake/blob/master/Modules%2FFindCUDA.cmake#L1028-1038). Apparently, after CUDA 3.0, the C headers were moved under the C directory. If you know this is wrong, please let me know--I am not familiar with CUDA, I just tried to replicate the FindCUDA search order.

Reply:
I'm not sure it makes sense to include `NVSDKCOMPUTE_ROOT`; if it's empty, the effect is to consider `/C` which probably isn't what we want.

I'd suggest we simplify this so it just checks that the user has specified (a reasonable value for) `OMR_CUDA_HOME` if `OMR_OPT_CUDA` is set.

Reply:
Or make it behave like the documentation for `OMR_OPT_CUDA` says:

> Path to the CUDA SDK. Takes precedence over CUDA_HOME in OMR

Reply:
I think it's reasonable to try to automatically locate CUDA. I would rather improve the search path than remove the functionality outright.

Reply:
I tried a cmake build on zLinux yesterday: the search for CUDA failed. CMake is configured _explicitly_ with `-DJ9VM_OPT_CUDA=OFF` which OpenJ9 maps to `OMR_OPT_CUDA=OFF`. We shouldn't even be _trying_ to find CUDA in that case. I don't know if there's a way to tell the difference between OFF by default and OFF on the command-line: if not, then I suggest we have to scale this back so only validate a path supplied in `CUDA_BIN_PATH` (or whatever CMake variables we settle on).

Reply:
Yes, with this PR, cmake will search for CUDA only if `OMR_OPT_CUDA` is enabled. We can still provide a default search path for CUDA headers, and there is no need to differentiate why the feature is off. Does that sound good?

Reply:
If the caller specifies `OMR_CUDA_HOME`, this should not consider any other location that might happen to be an acceptable CUDA installation.

Reply:
Sounds good to me 👍